### PR TITLE
Player 1380

### PR DESCRIPTION
**Pull request by dallen42**

### CODE REVIEW

File: js/utils/constants.js [hunk 1]
```diff
@@ -918,6 +918,115 @@
...
+       *   <li>The ad type (Linear/NonLinear) (Overlay/Video).</li>
+       * </ul>
+       * @event OO.EVENTS#AD_SDK_IMPRESSION
+       */
+      AD_SDK_IMPRESSION: 'adSdkImpression',
+
+      /**
+       * This event is triggered when an ad has completed playback.        
+       * <ul>
+       *   <li>The ad plugin that failed to load.</li>
```
Comment (aeng7) [inline js/utils/constants.js]: This should probably be "the ad plugin" and not the ad plugin that failed to load.

File: js/utils/constants.js [hunk 2]
```diff
@@ -918,6 +918,115 @@
...
+       *   <li>The media file URL.</li>
+       * </ul>
+       * @event OO.EVENTS#AD_PLAYBACK_ERROR
+       */
+      AD_PLAYBACK_ERROR: 'adPlaybackError',
+
+      /**
+       * This event is triggered when the ad plugin sdk records an impression event.        
+       * <ul>
+       *   <li>The ad plugin that failed to load.</li>
```
Comment (aeng7) [inline js/utils/constants.js]: This should probably be "the ad plugin" and not the ad plugin that failed to load.

File: js/utils/constants.js [hunk 3]
```diff
@@ -918,6 +918,115 @@
...
+       *   <li>The time the ad was scheduled to play.</li>
+       *   <li>The ad load time - time between ad request success and first frame started.</li>
+       *   <li>The ad protocol (VAST/VPAID).</li>
+       *   <li>The ad type (Linear/NonLinear) (Overlay/Video).</li>
```
Comment (aeng7) [inline js/utils/constants.js]: Is this an object with 2 parameters?

**Comment by aeng7** [comment]
If changes are made, please make sure to update the mjolnir copy as well.

File: js/utils/constants.js (hunk 3 above)
Comment (dallen42) [reply js/utils/constants.js]: No, I just put that as shorthand to represent all 4 possible states. LinearOverlay, NonLinearOverlay, LinearVideo, and NonLinearVideo

File: js/utils/constants.js (hunk 1 above)
Comment (dallen42) [reply js/utils/constants.js]: Fixed

File: js/utils/constants.js (hunk 2 above)
Comment (dallen42) [reply js/utils/constants.js]: Fixed

File: js/utils/constants.js (hunk 3 above)
Comment (dallen42) [reply js/utils/constants.js]: @aeng7 clarified this in a new commit

File: js/utils/constants.js [hunk 4]
```diff
@@ -918,6 +918,115 @@
        */
       VC_TAG_FOUND: 'videoTagFound',
 
+      /**
+       * This event is triggered when an ad sdk has been loaded successfully. the handler is called with:
```
Comment (pilievOoyala) [inline js/utils/constants.js]: The 't' in second sentence needs to be capitialized.

File: js/utils/constants.js [hunk 5]
```diff
@@ -918,6 +918,115 @@
...
+       * <ul>
+       *   <li>The ad plugin loaded.</li>
+       * </ul>
+       * @event OO.EVENTS#AD_SDK_LOADED
+       */
+      AD_SDK_LOADED: 'adSdkLoaded',
+
+      /**
+       * This event is triggered when there is an failure to load the ad sdk.
```
Comment (pilievOoyala) [inline js/utils/constants.js]: For consistency, can you add in 'The handler is called with the following arguments:' to all the new messages below.

File: js/utils/constants.js (hunk 4 above)
Comment (dallen42) [reply js/utils/constants.js]: Fixed

File: js/utils/constants.js (hunk 5 above)
Comment (dallen42) [reply js/utils/constants.js]: Fixed

**Pull request merged**